src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const persons = document.body.querySelectorAll('li');
+
+function sortSalary() {
+  let previous = persons[0]
+    .getAttribute('data-salary')
+    .replace((/[,$]/g, ''), 10);
+
+  for (let i = 1; i < persons.length; i++) {
+    const current = persons[i]
+      .getAttribute('data-salary')
+      .replace((/[,$]/g, ''), 10);
+
+    if (current > previous) {
+      persons[i - 1].before(persons[i]);
+    }
+
+    previous = current;
```

Comment:
The sorting logic is incorrect. The current implementation only swaps adjacent elements if the current salary is greater than the previous one, which does not result in a sorted list. Consider using a more appropriate sorting algorithm, such as bubble sort or array sort method.